### PR TITLE
mod+lightning+chainkit: update to lnd v0.17.1-beta.rc1 and expose GetBlockHeader

### PR DESCRIPTION
**Pull request by jharveyb**
#### Pull Request Checklist

Only bumping the lnd version to `v0.17.1-beta.rc1` and exposing the `GetBlockHeader` call.

IIUC this should go in a new branch, not master.

- [ ] PR is opened against correct version branch.
- [ ] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [ ] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

### CODE REVIEW

**Comment by bhandras** [comment]
Could you lease open against this branch: https://github.com/lightninglabs/lndclient/tree/lnd-17-0

File: chainkit_client.go [hunk 1]
```diff
@@ -77,6 +82,33 @@ func (s *chainKitClient) GetBlock(ctxParent context.Context,
 	return msgBlock, nil
 }
 
+// GetBlockHeader returns a block header given the corresponding block hash.
+func (s *chainKitClient) GetBlockHeader(ctxParent context.Context,
```
Comment (bhandras) [inline chainkit_client.go:86]: I believe you also need to add this to `testdata/permissions.json`.

**Review by bhandras — APPROVED** [review]
LGTM, pending changing target branch to `lnd-17-0`.

File: chainkit_client.go (hunk 1 above)
Comment (jharveyb) [reply chainkit_client.go:86]: added all the other `ChainKit` methods.

**Pull request merged**